snapcraft/internal/elf.py

```diff
@@ -376,6 +377,16 @@ def load_dependencies(
                 library_paths.add(l.path)
         return library_paths
 
+    def is_dynamic_executable(self) -> bool:
+        # This way of answering the question is perhaps a bit OTT. But it works.
+        try:
+            ef = elftools.elf.elffile.ELFFile(open(self.path, "rb"))
+        except ELFError as elf_error:
+            logger.debug("ELFFile({}) failed: {}".format(self.path, elf_error))
+            return False
+        else:
+            return "PT_DYNAMIC" in [s.header.p_type for s in ef.iter_segments()]
```

Comment:
just for reference, we also have https://github.com/snapcore/snapcraft/pull/2519/files#diff-f7cf307285f042f35434bbe680606c77R258 and those are all set as properties here https://github.com/snapcore/snapcraft/pull/2519/files#diff-f7cf307285f042f35434bbe680606c77R219 so it might be good to just do it on load, we can leave as is for now if we have a bug for this and a code comment TODO with the LP: # here